socks5.py

```diff
@@ -371,7 +377,7 @@ def handle_connect(self, address, port):
         except socket.timeout:
             logging.info('%s: connection timed out', log_tag)
         except Exception as e:
-            if str(e) == "[Errno 54] Connection reset by peer":
```

Comment:
This should to be in a separate `except socket.error` (or `except OSError`) block, since only those classes are guaranteed to have an `errno` attribute. Alternatively, `getattr(e, "errno", None)` would work.